### DISCUSSION TITLE
500 Error on api route only on production

### DISCUSSION THREAD

**Discussion by lucasbaze** [blitz-js/blitz#2352] [Q&A]
👋  Having a strange issue with an api route not working in production on render.com, but working fine in development.

I suspect it may have something to do with `getInitialProps` on the server, but I'm not entirely sure how to properly diagnose this issue. Much appreciated in advance. 

So the file below `create-thought/index.tsx` makes a fetch request to `/api/uploads` to complete the upload process. I can confirm the below code works in development, but not on production. 

I've looked at https://stackoverflow.com/questions/60293703/netx-js-returns-500-internal-server-error-in-production that I suspect could the underlying issue, but I'm not sure.

`api/uploads/index.ts`
```
import { BlitzApiRequest, BlitzApiResponse } from "blitz"
import Mux from "@mux/mux-node"
const { Video } = new Mux(process.env.MUX_TOKEN_ID, process.env.MUX_TOKEN_SECRET)

const uploadHandler = async (req: BlitzApiRequest, res: BlitzApiResponse) => {
  const { method } = req

  switch (method) {
    case "POST":
      try {
        const upload = await Video.Uploads.create({
          new_asset_settings: { playback_policy: "public" },
          cors_origin: "https://thoughtful.video",
        })
        res.json({
          id: upload.id,
          url: upload.url,
        })
      } catch (e) {
        console.error("Request error", e)
        res.status(500).json({ error: "Error creating upload" })
      }
      break
    default:
      res.setHeader("Allow", ["POST"])
      res.status(405).end(`Method ${method} Not Allowed`)
  }
}

export default uploadHandler
```

`create-thought/index.ts`
```
const createUpload = async () => {
    try {
      console.log("Attempting to Upload to endpoint")
      const url = `${getClientHost()}/api/uploads`
      console.log("Url to Fetch: ", url)
      return fetch(url, {
        method: "POST"
      })
        .then((res) => res.json())
        .then(({ id, url }) => {
          setUploadId(id)
          return url
        })
    } catch (e) {
      console.error("Error in createUpload", e)
      setErrorMessage("Error creating upload")
    }
  }
```

`getClientHost` resolves to `https://thoughtful.video/api/uploads`

I've got a couple other api's including oauth to google and stripe endpoints that are all working as expected. 

![Screen Shot 2021-05-16 at 6 00 28 PM](https://user-images.githubusercontent.com/16271037/118417284-9fe35400-b670-11eb-9c46-a35111c40936.png)

Any help would be greatly appreciated!

**Comment by flybayer** [comment]
You aren't using `getInitialProps` so that's not in the code path here.

500 error means something is failing. So now you need to debug exactly what is failing. Add log statements and things to find the exact line that is broken. Then try to troubleshoot why that line is broken.
